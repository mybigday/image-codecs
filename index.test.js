/**
 * @jest-environment jsdom
 */

const { encode, decode } = require('./')

const emptyImage = {
  width: 1,
  height: 1,
  data: new Uint8Array([0, 0, 0, 0])
}

test('encode', () => {
  expect(encode(emptyImage, 'jpeg')).toMatchSnapshot()
  expect(encode(emptyImage, 'png')).toMatchSnapshot()
  expect(encode(emptyImage, 'bmp')).toMatchSnapshot()
})

test('decode', () => {
  expect(decode(encode(emptyImage, 'jpeg'))).toMatchSnapshot()
  expect(decode(encode(emptyImage, 'png'))).toMatchSnapshot()
  expect(decode(encode(emptyImage, 'bmp'))).toMatchSnapshot()
})
