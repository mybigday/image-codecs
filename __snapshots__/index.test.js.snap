// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decode 1`] = `
{
  "data": Uint8Array [
    0,
    0,
    0,
    255,
  ],
  "height": 1,
  "width": 1,
}
`;

exports[`decode 2`] = `
{
  "data": Uint8Array [
    0,
    0,
    0,
    0,
  ],
  "height": 1,
  "width": 1,
}
`;

exports[`decode 3`] = `
{
  "data": Uint8Array [
    0,
    0,
    0,
    0,
  ],
  "height": 1,
  "width": 1,
}
`;

exports[`encode 1`] = `
{
  "data": [
    255,
    216,
    255,
    224,
    0,
    16,
    74,
    70,
    73,
    70,
    0,
    1,
    1,
    0,
    0,
    1,
    0,
    1,
    0,
    0,
    255,
    219,
    0,
    132,
    0,
    16,
    11,
    12,
    14,
    12,
    10,
    16,
    14,
    13,
    14,
    18,
    17,
    16,
    19,
    24,
    40,
    26,
    24,
    22,
    22,
    24,
    49,
    35,
    37,
    29,
    40,
    58,
    51,
    61,
    60,
    57,
    51,
    56,
    55,
    64,
    72,
    92,
    78,
    64,
    68,
    87,
    69,
    55,
    56,
    80,
    109,
    81,
    87,
    95,
    98,
    103,
    104,
    103,
    62,
    77,
    113,
    121,
    112,
    100,
    120,
    92,
    101,
    103,
    99,
    1,
    17,
    18,
    18,
    24,
    21,
    24,
    47,
    26,
    26,
    47,
    99,
    66,
    56,
    66,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    99,
    255,
    192,
    0,
    17,
    8,
    0,
    1,
    0,
    1,
    3,
    1,
    17,
    0,
    2,
    17,
    1,
    3,
    17,
    1,
    255,
    196,
    1,
    162,
    0,
    0,
    1,
    5,
    1,
    1,
    1,
    1,
    1,
    1,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    16,
    0,
    2,
    1,
    3,
    3,
    2,
    4,
    3,
    5,
    5,
    4,
    4,
    0,
    0,
    1,
    125,
    1,
    2,
    3,
    0,
    4,
    17,
    5,
    18,
    33,
    49,
    65,
    6,
    19,
    81,
    97,
    7,
    34,
    113,
    20,
    50,
    129,
    145,
    161,
    8,
    35,
    66,
    177,
    193,
    21,
    82,
    209,
    240,
    36,
    51,
    98,
    114,
    130,
    9,
    10,
    22,
    23,
    24,
    25,
    26,
    37,
    38,
    39,
    40,
    41,
    42,
    52,
    53,
    54,
    55,
    56,
    57,
    58,
    67,
    68,
    69,
    70,
    71,
    72,
    73,
    74,
    83,
    84,
    85,
    86,
    87,
    88,
    89,
    90,
    99,
    100,
    101,
    102,
    103,
    104,
    105,
    106,
    115,
    116,
    117,
    118,
    119,
    120,
    121,
    122,
    131,
    132,
    133,
    134,
    135,
    136,
    137,
    138,
    146,
    147,
    148,
    149,
    150,
    151,
    152,
    153,
    154,
    162,
    163,
    164,
    165,
    166,
    167,
    168,
    169,
    170,
    178,
    179,
    180,
    181,
    182,
    183,
    184,
    185,
    186,
    194,
    195,
    196,
    197,
    198,
    199,
    200,
    201,
    202,
    210,
    211,
    212,
    213,
    214,
    215,
    216,
    217,
    218,
    225,
    226,
    227,
    228,
    229,
    230,
    231,
    232,
    233,
    234,
    241,
    242,
    243,
    244,
    245,
    246,
    247,
    248,
    249,
    250,
    1,
    0,
    3,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    0,
    0,
    0,
    0,
    0,
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
    11,
    17,
    0,
    2,
    1,
    2,
    4,
    4,
    3,
    4,
    7,
    5,
    4,
    4,
    0,
    1,
    2,
    119,
    0,
    1,
    2,
    3,
    17,
    4,
    5,
    33,
    49,
    6,
    18,
    65,
    81,
    7,
    97,
    113,
    19,
    34,
    50,
    129,
    8,
    20,
    66,
    145,
    161,
    177,
    193,
    9,
    35,
    51,
    82,
    240,
    21,
    98,
    114,
    209,
    10,
    22,
    36,
    52,
    225,
    37,
    241,
    23,
    24,
    25,
    26,
    38,
    39,
    40,
    41,
    42,
    53,
    54,
    55,
    56,
    57,
    58,
    67,
    68,
    69,
    70,
    71,
    72,
    73,
    74,
    83,
    84,
    85,
    86,
    87,
    88,
    89,
    90,
    99,
    100,
    101,
    102,
    103,
    104,
    105,
    106,
    115,
    116,
    117,
    118,
    119,
    120,
    121,
    122,
    130,
    131,
    132,
    133,
    134,
    135,
    136,
    137,
    138,
    146,
    147,
    148,
    149,
    150,
    151,
    152,
    153,
    154,
    162,
    163,
    164,
    165,
    166,
    167,
    168,
    169,
    170,
    178,
    179,
    180,
    181,
    182,
    183,
    184,
    185,
    186,
    194,
    195,
    196,
    197,
    198,
    199,
    200,
    201,
    202,
    210,
    211,
    212,
    213,
    214,
    215,
    216,
    217,
    218,
    226,
    227,
    228,
    229,
    230,
    231,
    232,
    233,
    234,
    242,
    243,
    244,
    245,
    246,
    247,
    248,
    249,
    250,
    255,
    218,
    0,
    12,
    3,
    1,
    0,
    2,
    17,
    3,
    17,
    0,
    63,
    0,
    243,
    250,
    0,
    255,
    0,
    255,
    217,
  ],
  "type": "Buffer",
}
`;

exports[`encode 2`] = `
Uint8Array [
  137,
  80,
  78,
  71,
  13,
  10,
  26,
  10,
  0,
  0,
  0,
  13,
  73,
  72,
  68,
  82,
  0,
  0,
  0,
  1,
  0,
  0,
  0,
  1,
  8,
  6,
  0,
  0,
  0,
  31,
  21,
  196,
  137,
  0,
  0,
  0,
  11,
  73,
  68,
  65,
  84,
  120,
  94,
  99,
  96,
  0,
  2,
  0,
  0,
  5,
  0,
  1,
  6,
  20,
  117,
  163,
  0,
  0,
  0,
  0,
  73,
  69,
  78,
  68,
  174,
  66,
  96,
  130,
]
`;

exports[`encode 3`] = `
{
  "data": [
    66,
    77,
    58,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    54,
    0,
    0,
    0,
    40,
    0,
    0,
    0,
    1,
    0,
    0,
    0,
    255,
    255,
    255,
    255,
    1,
    0,
    24,
    0,
    0,
    0,
    0,
    0,
    4,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "type": "Buffer",
}
`;
